### PR TITLE
Add a page for the contract developer position

### PR DESCRIPTION
**Pull request by ibennetch**
- [x] Perhaps we should have Conservancy review this before publishing.
- [x] I was uncertain about the use of 'pmaweb/urls.py' and 'pmaweb/views.py' so I kind of guessed at what is right.
- [x] The text is copied directly from the [January posting](https://www.phpmyadmin.net/news/2017/1/25/seeking-phpmyadmin-developer-one-year-contract-position/).
- [ ] Should we make this more prominent (perhaps even to the point of including it as a green sidebar link)?

Signed-off-by: Isaac Bennetch <bennetch@gmail.com>

### CODE REVIEW

File: pmaweb/views.py [hunk 1]
```diff
@@ -50,6 +50,7 @@
     'about': 'about',
     'awards': 'awards',
     'contest': 'contest',
+    'contractor': 'contractor',
```
Comment (nijel) [inline pmaweb/views.py]: This is really just a compatibility mapping of old website URLs, no need to add anything here. I've improved the docs in 5294675.

File: pmaweb/templates/contractor.html [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+<p>
+The ideal candidate will dynamically balance their workload based on outstanding issues and priority, but is
+anticipated to break down to these percentages (in priority order of what's most important to least important):
+</p>
+
+<ul>
+  <li>Security maintenance (5%)</li>
+  <li>Bug fixing and issue assessment (45%)</li>
+  <li>Code base improvement like refactoring and writing unit tests (45%)</li>
+  <li>Implementation of new features (5%)</li>
```
Comment (nijel) [inline pmaweb/templates/contractor.html]: I'm not really sure we should put the percents here - I think it's more reasonable to adjust workload based on current situation. It never really worked for me even though we were adjusting it for my second contract.

What I think is important the priority of things, what is already covered.

Another thing comes to my mind - do we require all contractors to work on security issues?

File: pmaweb/templates/contractor.html [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+</ul>
+
+<h3>Work conditions</h3>
+<ul>
+  <li>The Developer shall publish a weekly blog post about his/her work to the phpMyAdmin community.</li>
+  <li>The Developer shall send a monthly invoice to the Software Freedom Conservancy for the work done during that month.</li>
```
Comment (nijel) [inline pmaweb/templates/contractor.html]: I think it should state "invoice and report". At least that's the practice so far, even though the contracts maybe don't mention that...

File: pmaweb/templates/contractor.html [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+<p>
+The candidate will be able to demonstrate a very good knowledge of phpMyAdmin's code base. In addition, we expect
+excellent skills in all of the technologies used by phpMyAdmin (PHP, HTML, JavaScript, jQuery, CSS, MySQL) and excellent communication skills.
+</p>
+
+<h3>Applying</h3>
+
+<p>
+Candidates should submit their proposal (including their CV, availability, and financial terms) to: contract@phpmyadmin.net.
```
Comment (nijel) [inline pmaweb/templates/contractor.html]: Shouldn't there be "proposals"? (I'm not really sure, but I'd write it that way, but you're usually better in English than me ;-)).

File: pmaweb/urls.py [hunk 5]
```diff
@@ -149,6 +149,14 @@
         name='contribute'
     ),
     url(
+        r'^contractor/$',
+        PMAView_as_view_(
```
Comment (nijel) [inline pmaweb/urls.py]: Should be "PMAView.as_view"

File: pmaweb/templates/contractor.html (hunk 2 above)
Comment (ibennetch) [inline pmaweb/templates/contractor.html]: Ok, see comments seem reasonable. I've adjusted the text.

I don't have a strong feeling about the security issue aspect. I've removed it for now.

File: pmaweb/templates/contractor.html (hunk 3 above)
Comment (ibennetch) [inline pmaweb/templates/contractor.html]: Yes, this is a good change, although the actual contracts do specify that a report is expected. I've added it here.

**Comment by nijel** [comment]
BTW: I've talked with Bradley about this and he sees no problem in doing it this way.

**Comment by mauriciofauth** [comment]
I think it would be interesting to talk about contractors developers in the _Fund our project_ section in _Contribute_ page.

**Comment by nijel** [comment]
It certainly makes sense to mention it there.

**Comment by ibennetch** [comment]
I've also added the developer to the donation section, any comment about that?

File: pmaweb/templates/contribute.html [hunk 6]
```diff
@@ -87,4 +91,9 @@ <h3 id="devel">Developing</h3>
 you help us in this area and possibly increase our presence at conferences.
 </p>
 
+<p>
+We also use donated funds to hire <a href="{% url 'contractor' %}">contract developers</a> who
```
Comment (nijel) [inline pmaweb/templates/contribute.html]: This URL name should match name used in urls.py here.

**Comment by nijel** [comment]
I've made some fixes here: https://github.com/ibennetch/pma-website/pull/1

**Comment by ibennetch** [comment]
I think this is ready to go.

**Comment by nijel** [comment]
I agree, merged!

**Comment by ibennetch** [comment]
Thanks for the help

**Comment by nijel** [comment]
JFYI: It's live here: https://www.phpmyadmin.net/contractor/

File: pmaweb/templates/contractor.html [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+  <li>Bug fixing and issue assessment</li>
+  <li>Code base improvement like refactoring and writing unit tests</li>
+  <li>Implementation of new features</li>
+</ul>
+
+<h3>Work conditions</h3>
+<ul>
+  <li>The Developer shall publish a weekly blog post about his/her work to the phpMyAdmin community.</li>
+  <li>The Developer shall send a monthly invoice and report to the Software Freedom Conservancy for the work done during that month.</li>
+  <li>All work produced by the Developer is to be licensed under "GPLv2 or later".</li>
```
Comment (mauriciofauth) [inline pmaweb/templates/contractor.html]: Is the license not just _GPLv2_ instead of _GPLv2 or later_?

Comment (nijel) [inline pmaweb/templates/contractor.html]: The license is currently GPLv2, but we want all contributions to be GPLv2 or later to allow relicensing in the future, see also https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO

Comment (mauriciofauth) [inline pmaweb/templates/contractor.html]: I was a bit confused about that. Thank you for the explanation.

**Pull request merged**